examples/ecs/iter_combinations_simple.rs

```diff
@@ -0,0 +1,57 @@
+//! A simple example showing how iter_combinations works
+
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_startup_system(add_entities)
+        .add_system(example)
+        .run();
+}
+
+#[derive(Component)]
+struct A(usize);
+
+fn add_entities(mut commands: Commands) {
+    commands.spawn(A(1));
+    commands.spawn(A(2));
+    commands.spawn(A(3));
+    commands.spawn(A(4));
+}
+
+// Below is a simple example of iter_combinations.
+// The iter_combinations method produces the set of combinations of
+// components without repeats.
+
+// The iter combintions method is useful when there are a set of entities with components that all need to interact with
+// eachother in some way. Since iter_combinations does not repeat entities, you can call it without worrying about entities
+// interacting with themselves.
+fn example(query: Query<&A>) {
+    // The below loop will print out...
+    // [1, 2], [1, 3], [1, 4], [2, 3], [2, 4], [3, 4]
```

Comment:
Sorry, I can't see where it is doing the printing? Also, would the printing happen in order?

Reply:
I removed the print statements, but that was what it was consistently printing out. So it is probably deterministic. 

I can change the sentence to be something like 

"The order for the items in the array will be as follows".